### PR TITLE
Application skip ahead page

### LINKED ISSUE #6063: Special view if you skipped ahead to a disabled step somehow
## ✨ Feature

Error page when the user advanced to a  disabled step by manually entering url 

## 🕵️ Details

User should be allowed to skip ahead without completing the necessary steps. New error page should be added to be shown in this scenario.

## 🎨 Design File

https://www.figma.com/file/BXYVsu1fo9LS2IySJDUINY/Applicants---Job-application?node-id=636-59839&t=JufuhV0iNsTnIOxc-0
## 📸 Screenshot

![image](https://user-images.githubusercontent.com/89482910/227228967-5a0da0f4-a1e1-411e-b3dd-efc1083c7f1b.png)

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

| EN                                                                                                                                                                                                                                                                                                  | FR                                                                                                                                                                                                                                                                                                                                                         | Desc                                  |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------|
| Uh oh, it looks like you jumped ahead!                                                                                                                                                                                                                                                              | Oh! on dirait que vous avez été trop vite!                                                                                                                                                                                                                                                                                                                 | Application step skipped page title   |
| The application process relies on collecting and reviewing information in a linear fashion. As a result, it’s not possible to complete this step just yet. The last step you were working on is highlighted in the steps area of the page, or you can use the button provided to go there directly. | Le processus de demande repose sur la collecte et l’examen des renseignements de manière linéaire. Par conséquent, il n’est pas encore possible de passer à cette étape. La dernière étape sur laquelle vous travailliez est mise en surbrillance dans la zone des étapes de la page, ou vous pouvez utiliser le bouton fourni pour y accéder directement. | Application step skipped page details |
|                                                                                                                                                                                                                                                                                                     |                                                                                                                                                                                                                                                                                                                                                            |                                       |

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Error page shown when the user skips ahead 
- [ ] **Return to the last step I was working on** button should take the user to step his last submitted step
- [ ] Logic for ensuring the stepper shows the correct state

## 🛑 Blockers

Blocked by {issue numbers}

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #6063 

## 👋 Introduction

This branch adds a "skip ahead" page if you manually navigate to a disabled step in the flow and updates the stepper status to reflect submitted steps.

~Bonus: it adds the missing nginx config for the new application pages.~ I'll let #6171 handle this.

## 🕵️ Details

I had to do a bit of refactoring of the stepper component to make this work.  Specifically, I removed the ability of the stepper to disable its own steps and instead rely on the parent component.  This was needed because only the parent component knows the submitted steps and which ones to unlock.  I also added a "default" step state for steps that are not completed, not current, but need to be clickable for the user to go back to if they jumped ahead.  It also fixes a bug where the success page highlighted the "Welcome" step as current.

## 🧪 Testing

New page routing at: https://github.com/GCTC-NTGC/gc-digital-talent/blob/4301e14957ce079fcc5cc6844015ba4d71bf2371/apps/web/src/components/Router.tsx#L1056 

1. Rebuild the app
2. Start a new application or find one already started. Note the value of `submitted_steps` for that application.
3. Manually navigate to the new application pages and manually update the submitted_steps for the application.
4. Confirm it works properly:
    - Already submitted pages should show normally (currently blank with no content)
    - Not submitted pages show the "oh no" page
    - The "oh no" page has a link back to the right spot 
    - Stepper shows submitted steps (error state is #6145)
    - Stepper component disables unsubmitted steps unless it is the next one to complete

## 📸 Screenshot

image

### CODE REVIEW

**Review by esizer — CHANGES_REQUESTED** [review]
WOW, this is super clean and great 🤯 

One _tiny_ a11y suggestion 😸

File: apps/web/src/pages/Applications/StepDisabledPage/StepDisabledPage.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          description: "Application step skipped page details",
+        })}
+      </p>
+      <hr
+        data-h2-background-color="base(gray.lighter)"
+        data-h2-height="base(1px)"
+        data-h2-width="base(100%)"
+        data-h2-border="base(none)"
+        data-h2-margin="base(x2, 0)"
+      />
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Applications/StepDisabledPage/StepDisabledPage.tsx]: Thanks!  I've never used it before - good to know.
[Use Separator component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6181/commits/8d4054c10fbfd5dd187f0fdf2330b05540c94654)

File: apps/web/src/lang/fr.json [hunk 2]
```diff
@@ -7372,5 +7372,13 @@
   "wMp4x6": {
     "defaultMessage": "Télécharger le guide (EN)",
     "description": "Link text for English guidance resource download"
+  },
+  "9vQj8o": {
+    "defaultMessage": "Oh! on dirait que vous avez été trop vite!",
```
Comment (patcon) [inline apps/web/src/lang/fr.json:7377]: Sorry, this looked a little funny, and while not a sure thing, seems the lowercase is not common unless a comma?
https://context.reverso.net/translation/french-english/Oh!+on+dirait+que

Comment (patcon) [reply apps/web/src/lang/fr.json:7377]: Happy to see this marked out of scope as a concern, tho!

**Comment by patcon** [comment]
started to review, but realzied i need to break for dinner before demo. didn't want to block, but i'll pick it up again if it's still here after demo :)

File: apps/web/src/lang/fr.json (hunk 2 above)
Comment (petertgiles) [reply apps/web/src/lang/fr.json:7377]: @gobyrne thought's on this one?

**Pull request merged**